collector/test/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+# Setup testing
+enable_testing()
+
+# Unit Tests
+file(GLOB TEST_SRC_FILES ${PROJECT_SOURCE_DIR}/test/*.cpp)
+foreach(test_file ${TEST_SRC_FILES})
+    get_filename_component(test_name ${test_file} NAME_WE)
+    add_executable("${test_name}" "${test_file}")
+
+    if(NOT DISABLE_PROFILING AND NOT USE_VALGRIND)
+        target_link_libraries(${test_name} profiler tcmalloc)
```

Comment:
Not sure I get it with `NOT USE_VALGRIND`: so, if we don't use Valgrind, we still link with tcmalloc? I think originally it was `NOT DEFINED ENV{USE_VALGRIND}`?

Reply:
Yeah, this condition is quite confusing. The end goal is that if we ARE using valgrind we don't want to link with gperftools because tcmalloc messes with it. I think what we really want is if `USE_VALGRIND` is `true` then we need to ensure `DISABLE_PROFILING` is also `true`, I have a similar change on a separate PR (too much duplicated efforts, sorry!).